### ISSUE TITLE
Repeated fields in parameter are set as a single string v.s. a slice of string (incompatible with `grpc-gateway`)

### ISSUE THREAD

**Issue by HuairuoYang** [googleapis/gapic-generator-go#1160]
I am using this client generator to generate clients that can talk to my server via  REST/GRPC in an easy approach. And I am using `grpc-gateway` for routing the HTTP/REST requests to GRPC. I thought both this and `grpc-gateway` are popular tools and they all generate code from protobuf so that they should be compatible with each other.
However I noticed that the repeated field parameter didn't work well.

Could you please help to check if this is a bug/issue, or if this is a feature that is nice to have?
I think people would be more than happy if the generated REST Clients can be directly used against `grpc-gateway`!

Thanks!

#### Environment details

  - Programming language: Golang
  - OS: MacOS
  - Language runtime version: v1.17.13
  - Package version:

#### Steps to reproduce
1. 
My proto is similar to this:
```proto
rpc List(ListRequest) returns (ListResponse) {
    option (google.api.http) = {
      get: "/v1/{parent=*}/lists"
    };
}

message ListRequest {
    enum Types {
        A=0;
        B=1;
    }
    repeated Types = 1; 
}
```
So I can send a request with Types = [0, 1].

2. 
And in the generated REST client's `List` method, the `Types` is set in this way
```
		if req.GetTypes() != nil {
			params.Add("types", fmt.Sprintf("%v", req.GetTypes()))
		}
```
This sets a single string `[0,1]` to the parameter `types`.

3. 
Whereas in the server side, `grpc-gateway` receives the request and parses the parameter using [this method: Parse](https://github.com/grpc-ecosystem/grpc-gateway/blob/v1/runtime/query.go#L37), where the input `values` is the `request.Form` which contains the parameter k-v pair.

And it will parse the above `types: "[0,1]"` to `key: types`, `value: "[0,1]"` (Note, it is a string `"[0,1]"`). And it will fail saying that `[0,1]` is not a correct enum type.

4. 
My solution is to adjust the above code in the generated REST clients:
```
		if req.GetTypes() != nil {
			for _,n := range req.GetTypes(){
				params.Add("types", fmt.Sprintf("%v", n))
			}
		}
```
to add multiple strings for the types and this will work perfectly with `grpc-gateway`.
This will set the parameter fields with a slice of strings `0`, `1` and `grpc-gateway` will parse and return correct values.

**Comment by noahdietz** [comment]
Thanks for the bug report. I think you are right and your suggested fix to the generated code seems like it would work. I will see about fixing this soon.

**Comment by noahdietz** [comment]
Relevant gRPC-HTTP Transcoding semantics that the generator is currently violating are documented in the following comments on the `google.api.http` message:
* https://github.com/googleapis/googleapis/blob/cb8c91471c9c4540aba3b684b1126e3d06b1ef0b/google/api/http.proto#L119-L120
* https://github.com/googleapis/googleapis/blob/cb8c91471c9c4540aba3b684b1126e3d06b1ef0b/google/api/http.proto#L215-L218

This is the standard for the behavior that @HuairuoYang sees in `grpc-gateway` and that we need to comply with in this code generator.

**Comment by HuairuoYang** [comment]
Thank you @noahdietz ! Could I ask when would this be fixed and which will it be included in the next release (when)?

**Comment by noahdietz** [comment]
> Thank you @noahdietz ! Could I ask when would this be fixed and which will it be included in the next release (when)?

Just opened a PR :) If merged today, I can release today. Otherwise, it will go out on Monday (we try not to release on Friday).

**Comment by HuairuoYang** [comment]
Impressive! Thanks!

**Comment by noahdietz** [comment]
Release to come on Monday @HuairuoYang

**Comment by noahdietz** [comment]
Should be available in `v0.33.2`: https://github.com/googleapis/gapic-generator-go/pull/1162

**Comment by HuairuoYang** [comment]
Verified that it works! Thanks again for fast response and execution of the team! @noahdietz
